### ISSUE TITLE
Class having render() and returning JSX no longer recognised as a component

### ISSUE THREAD

**Issue by worklez** [yannickcr/eslint-plugin-react#280]
This was the old behaviour, but has changed recently (#237 seems relevant). When there is a class that extends from component other than `Component` or `React.Component`, e.g.

```js
class Button extends PureComponent {
  render() {
    const {text} = this.props;
    return <div>{text}</div>;
  }
}

Button.propTypes = {
  text: React.PropTypes.string.isRequired
};
```

it wouldn't be recognised as component and prop types wouldn't be validated.

**Comment by yannickcr** [comment]
Yes, there was a few changes about component detection. There is still a few bugs (#267, #277) and I'm working on it.

**Comment by yannickcr** [comment]
Should be fixed by 2749a0b15bdc7747fe30cdebfde7f54cd41ba75e in `v3.7.0`. Let me know if it's not the case.
